oat-sa/tao-core-sdk-fe
# Fix/ADF-1166/ESlint errors in core/moduleLoader, also fix a wrong use of dynamic import

### Related to: https://oat-sa.atlassian.net/browse/ADF-1166

### Summary

Fix the ESLint error in `core/moduleLoader`.
Fix the wrong use of the dynamic import which was not returning anything (look at the changed code, it was surely not properly working).

**Note:** as the changes are pretty heavy, I split them all into several pull requests.

### Details

This pull request only targets the `core/moduleoader` part of the library.
The full picture can be seen from the main branch which contains all the fixes: https://github.com/oat-sa/tao-core-sdk-fe.git#fix/ADF-1166/eslint-errors

What has been made:
- replace `arguments` with the rest operator
- replace `var` with `const` or `let` (but not in all places, only in the touched files when it was not adding too much noise)
- apply object shorthand instead of duplicated function names
- apply arrow functions here and there when it helped, but not systematically to reduce the noise
- fix the wrong usage of dynamic import as its return value was not properly managed
- remove ESLint complain regarding the dynamic import

### How to test
- checkout and install the source code `npm i`
- run the linter: `npm run lint:src`, errors and warnings coming from the `core/moduleLoader` folder should have gone.
- run the linter: `npm run lint:test`, errors and warnings coming from the `core/moduleLoader` folder should have gone.
- run the unit tests: `npm run test core`
- install with other repositories, using the following scheme: `"@oat-sa/tao-core-sdk": "https://github.com/oat-sa/tao-core-sdk-fe.git#fix/ADF-1166/eslint-errors",`, then checking the unit tests and playing with TAO
